### ISSUE TITLE
c_linux_dbg_native.bins/dbg/h2_uds_test bad_ping GRPC_POLL_STRATEGY=epoll 

### ISSUE THREAD

**Issue by MakMukhi** [grpc/grpc#10894]
Test flake seen: https://grpc-testing.appspot.com/job/gRPC_pull_requests_linux/4984/testReport/junit/(root)/c_linux_dbg_native/bins_dbg_h2_uds_test_bad_ping_GRPC_POLL_STRATEGY_epoll/

**Comment by markdroth** [comment]
Similar failure seen in h2_http_proxy_test fixture:

https://grpc-testing.appspot.com/job/gRPC_master_asanitizer_c/1529/testReport/junit/(root)/c_linux_asan_native/bins_asan_h2_http_proxy_nosec_test_bad_ping_GRPC_POLL_STRATEGY_poll_cv/

**Comment by markdroth** [comment]
Yuchen, would you be willing to take a look at this?

**Comment by matt-kwong** [comment]
Same issue brought up by Mark seen here:

https://grpc-testing.appspot.com/job/gRPC_master_asanitizer_c/1533/testReport/junit/(root)/c_linux_asan_native/bins_asan_h2_http_proxy_nosec_test_bad_ping_GRPC_POLL_STRATEGY_poll_cv/

**Comment by murgatroid99** [comment]
Seen again in asan: https://grpc-testing.appspot.com/job/gRPC_master_asanitizer_c/1538/testReport/junit/(root)/c_linux_asan_native/bins_asan_h2_uds_test_bad_ping_GRPC_POLL_STRATEGY_epollsig/

**Comment by murgatroid99** [comment]
Also seen in asan with a different output: https://grpc-testing.appspot.com/job/gRPC_master_asanitizer_c/1546/testReport/junit/(root)/c_linux_asan_native/bins_asan_h2_http_proxy_nosec_test_bad_ping_GRPC_POLL_STRATEGY_poll_cv/

```
D0514 04:19:54.555806552    3276 test_config.c:403]          test slowdown factor: sanitizer=3, fixture=1, poller=5, total=15
D0514 04:19:54.556277478    3276 ev_posix.c:121]             Using polling engine: poll-cv
D0514 04:19:54.556359745    3276 dns_resolver.c:316]         Using native dns resolver
I0514 04:19:54.654817776    3276 http_proxy_fixture.c:460]   Proxy address: localhost:24100
I0514 04:19:54.712831685    3276 http_connect_handshaker.c:319] Connecting to server localhost:8573 via HTTP proxy ipv6:[::1]:24100
D0514 04:19:54.730904461    3651 http_proxy_fixture.c:328]   on_read_request_done: 0x62300000e100 "No Error"
D0514 04:19:54.865215526    3276 chttp2_transport.c:1576]    PING strike
I0514 04:19:54.896924704    3651 http_proxy_fixture.c:145]   HTTP proxy server read: {"created":"@1494735594.896839754","description":"Socket closed","fd":9,"file":"src/core/lib/iomgr/tcp_posix.c","file_line":293,"target_address":"ipv6:[::1]:8573"}
E0514 04:19:54.896985644    3276 bad_ping.c:215]             assertion failed: status == GRPC_STATUS_UNAVAILABLE



*******************************
Caught signal SIGABRT
I0514 04:19:54.897180429    3651 http_proxy_fixture.c:145]   HTTP proxy client read: {"created":"@1494735594.896971317","description":"FD shutdown","file":"src/core/lib/iomgr/ev_poll_posix.c","file_line":456,"referenced_errors":[{"created":"@1494735594.896839754","description":"Socket closed","fd":9,"file":"src/core/lib/iomgr/tcp_posix.c","file_line":293,"target_address":"ipv6:[::1]:8573"}]}
D0514 04:19:54.897208283    3651 http_proxy_fixture.c:117]   endpoints: 0x61600001fb80 0x61600001f280
bins/asan/h2_http_proxy_nosec_test(backtrace+0x41)[0x44ad11]
bins/asan/h2_http_proxy_nosec_test[0x938565]
```

**Comment by murgatroid99** [comment]
Also seen on windows: https://grpc-testing.appspot.com/job/gRPC_master_windows/2546/testReport/junit/(root)/c_windows_opt_native/vsprojects_Release_h2_full_nosec_test_exe_bad_ping/

**Comment by muxi** [comment]
Still failing in master:
https://grpc-testing.appspot.com/job/gRPC_master_asanitizer_c/1581/testReport/(root)/c_linux_asan_native/bins_asan_h2_http_proxy_nosec_test_bad_ping_GRPC_POLL_STRATEGY_poll_cv/

**Comment by y-zeng** [comment]
This should be fixed by #11589

**Comment by apolcyn** [comment]
failure seen on `asan c++`, not a tsan failure but an abort

```
D0627 23:05:51.546301514   15326 dns_resolver.c:301]         Using native dns resolver
Note: Google Test filter = End2end/End2endTest.ClientCancelsRequestStream/1
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from End2end/End2endTest
[ RUN      ] End2end/End2endTest.ClientCancelsRequestStream/1
D0627 23:05:51.547556819   15326 end2end_test.cc:216]        TestScenario{use_proxy=false, credentials='INSECURE_CREDENTIALS'}
D0627 23:05:51.547773272   15326 ev_posix.c:110]             Using polling engine: poll
D0627 23:05:51.547825271   15326 dns_resolver.c:301]         Using native dns resolver
I0627 23:05:51.550499212   15326 server_builder.cc:254]      Synchronous server. Num CQs: 4, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10
D0627 23:05:51.550763583   15326 ev_posix.c:110]             Using polling engine: poll
D0627 23:05:51.550798307   15326 dns_resolver.c:301]         Using native dns resolver
E0627 23:05:51.561553443   15492 completion_queue.c:494]     assertion failed: !cqd->shutdown_called



*******************************
Caught signal SIGABRT
```

https://grpc-testing.appspot.com/job/gRPC_pull_requests_asan_cpp/5622/

pr: #11629
